libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/market-role-group.service.ts

```diff
@@ -22,24 +39,34 @@ import { EditableMarketRoleRow } from './dh-market-participant-actor-market-role
 @Injectable()
 export class MarketRoleGroupService {
   readonly groupRows = (rows: EditableMarketRoleRow[]) => {
-    const marketRolesMap = rows.reduce((map, row) => {
-      map.set(row.marketRole, [...(map.get(row.marketRole) || [])]);
-      if (row.gridArea)
-        map.set(row.marketRole, [
-          ...map.get(row.marketRole),
-          {
-            id: row.gridArea,
-            meteringPointTypes: row.meteringPointTypes,
-          },
-        ]);
-      return map;
-    }, new Map());
-
     const result: MarketRole[] = [];
 
-    marketRolesMap.forEach((v, k) =>
-      result.push({ marketRole: k, gridAreas: v })
-    );
+    for (let index = 0; index < rows.length; index++) {
```

Comment:
nitpick: SonarCloud flags this loop as a "code smell" https://sonarcloud.io/project/issues?id=geh-frontend-app&pullRequest=1114&resolved=false&types=CODE_SMELL

suggestion: consider using a `for-of` loop as they suggest.

Reply:
Fixed in https://github.com/Energinet-DataHub/greenforce-frontend/pull/1114/commits/4069d56e668a1cfeac17e111e7f1ab4d9d487175